tests/lib/resource_test.py

```diff
@@ -218,17 +218,34 @@ def test__invoke_handler_not_found(resource):
 
 
 def test__invoke_handler_was_found(resource):
-    mock_handler = resource.handler(Action.CREATE)(Mock(return_value=sentinel.response))
+    progress_event = Mock(
+        "cloudformation_cli_python_lib.interface.ProgressEvent", autospec=True
+    )()
+    mock_handler = resource.handler(Action.CREATE)(Mock(return_value=progress_event))
 
     resp = resource._invoke_handler(
         sentinel.session, sentinel.request, Action.CREATE, sentinel.context
     )
-    assert resp is sentinel.response
+    assert resp is progress_event
     mock_handler.assert_called_once_with(
         sentinel.session, sentinel.request, sentinel.context
     )
 
 
+@pytest.mark.parametrize("action", [Action.LIST, Action.READ])
+def test__invoke_handler_non_mutating_must_be_synchronous(resource, action):
+    progress_event = Mock(
```

Comment:
haven't dived in too much, but is there a reason to mock the progress event rather than just creating an actual progress event?